src/nuc/token.py

```diff
@@ -0,0 +1,240 @@
+"""
+NUC tokens.
+"""
+
+import json
+import re
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import Any, Dict, List
+
+from .policy import Policy
+
+_DID_PATTERN = re.compile("did:([a-zA-Z0-9_-]+):([a-zA-Z0-9]{66})")
+_HEX_PATTERN = re.compile("[a-zA-Z0-9]+")
```

Comment:
I _blindly_ trust you that this is correct.

Reply:
Given that nucs only use the 'nil' method currently, would it be better to fix the method rather than accept a regex?

Reply:
I'm ok with that; in the rust impl we also allow any method so I'll change it there as well.